src/Microsoft.Health.Fhir.TemplateManagement/Models/ImageInfo.cs

```diff
@@ -55,35 +56,67 @@ public static bool IsDefaultTemplateImageReference(string imageReference)
+        {
+            try
+            {
+                ValidateImageReference(imageReference);
+            }
+            catch
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        public static void ValidateImageReference(string imageReference)
         {
             var registryDelimiterPosition = imageReference.IndexOf(ImageRegistryDelimiter, StringComparison.InvariantCultureIgnoreCase);
             if (registryDelimiterPosition <= 0 || registryDelimiterPosition == imageReference.Length - 1)
             {
-                return false;
+                throw new ImageReferenceException(TemplateManagementErrorCode.InvalidReference, "Template reference format is invalid: registry server is missing.");
             }
             else
             {
                 imageReference = imageReference.Substring(registryDelimiterPosition + 1);
-
+                string imageName = imageReference;
                 if (imageReference.Contains(ImageDigestDelimiter, StringComparison.OrdinalIgnoreCase))
                 {
                     var digestDelimiterPosition = imageReference.IndexOf(ImageDigestDelimiter, StringComparison.InvariantCultureIgnoreCase);
                     if (digestDelimiterPosition <= 0 || digestDelimiterPosition == imageReference.Length - 1)
                     {
-                        return false;
+                        throw new ImageReferenceException(TemplateManagementErrorCode.InvalidReference, "Template reference format is invalid: digest is missing.");
+                    }
+                    else
+                    {
+                        imageName = imageReference.Substring(0, digestDelimiterPosition);
                     }
                 }
                 else if (imageReference.Contains(ImageTagDelimiter, StringComparison.OrdinalIgnoreCase))
                 {
                     var tagDelimiterPosition = imageReference.IndexOf(ImageTagDelimiter, StringComparison.InvariantCultureIgnoreCase);
                     if (tagDelimiterPosition <= 0 || tagDelimiterPosition == imageReference.Length - 1)
                     {
-                        return false;
+                        throw new ImageReferenceException(TemplateManagementErrorCode.InvalidReference, "Template reference format is invalid: tag is missing.");
+                    }
+                    else
+                    {
+                        imageName = imageReference.Substring(0, tagDelimiterPosition);
                     }
                 }
+
+                ValidateImageName(imageName);
             }
+        }
 
-            return true;
+        private static void ValidateImageName(string imageName)
+        {
+            // Reference docker's image name format: https://docs.docker.com/engine/reference/commandline/tag/#extended-description
+            Regex imageNameRegex = new Regex(@"^([a-z0-9])+((\.{1}|\-+|_{1,2})[a-z0-9]+)*$");
```

Comment:
We can make this regex a private/static member of this class to avoid creating object on each validation.
Besides, the regex can be ```^[a-z0-9]+([_\.|__|\-+][a-z0-9]+)*(\/[a-z0-9]+([_\.|__|\-+][a-z0-9]+)*)*$```
. [Reference](https://github.com/docker/distribution/blob/master/reference/reference.go)

Reply:
Test cases:   
- org/image-name
   - org/image-name
    - image-name
     - image.name
    - image__name
    - image-----name